### PR TITLE
Enable retrieving artifacts from Bintray

### PR DESCRIPTION
**Pull request by xnejp03**
To invoke this, services should drop this from their configs:

binary.groupId
binary.nexus

binary.artifact should lose the _2.10 or _2.11 suffix (and use the repo name as it is in bintray)

Users must set these env variables:

BINTRAY_USER = [bintray_username]
BINTRAY_PASS = [bintray_api_key]

### CODE REVIEW

File: servicemanager/service/smplayservice.py [hunk 1]
```diff
@@ -117,24 +119,38 @@ def start_from_binary(self):
                 print b.fail + "ERROR: could not start '" + self.service_name + "' " + b.endc
             return popen_output.pid
 
-    def _unzip_play_application(self):
+    def _unpack_play_application(self, extension):
```
Comment (rpowis) [inline servicemanager/service/smplayservice.py]: Can't we determine the extension automagically rather than pass it in?

Comment (xnejp03) [inline servicemanager/service/smplayservice.py]: We do, a few lines above, I think

Comment (rpowis) [inline servicemanager/service/smplayservice.py]: ah yes. kinda. I meant using [os.path.splitext](https://docs.python.org/2/library/os.path.html#os.path.splitext) to grab the extension.

I should've left this comment against [this line](https://github.com/hmrc/service-manager/pull/34/files#diff-66a883b6dfd002b95edaafaed230bcffR18) instead

Comment (xnejp03) [inline servicemanager/service/smplayservice.py]: We kinda need that, as the Play apps are packaged with .zip in Nexus and .tgz in Bintray - so to pick the right file, we have to tell it what to look for.

Comment (rpowis) [inline servicemanager/service/smplayservice.py]: Yes, we need to work out what to do based on the extension but my point is that we can infer the extension from the filename.

i.e. `get_play_app_extension ` could be
```
filename, extension = os.path.splitext(filename)
return extension
```

**Comment by rpowis** [comment]
We don't need to hit the Bintray API. We can grab the latest version from the `maven-metadata.xml` file in each artifact's directory. 

i.e. https://dl.bintray.com/hmrc/release-candidates/uk/gov/hmrc/help-frontend_2.11/maven-metadata.xml

```
<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>uk.gov.hmrc</groupId>
  <artifactId>help-frontend_2.11</artifactId>
  <version>1.26.0-3-gd7ed03c</version>
  <versioning>
    <latest>1.26.0-3-gd7ed03c</latest>
    <release>1.26.0-3-gd7ed03c</release>
    <versions>
      <version>1.26.0-1-gd0dba7c</version>
      <version>1.26.0-2-gd213a4f</version>
      <version>1.26.0-3-gd7ed03c</version>
    </versions>
    <lastUpdated>20150804143826</lastUpdated>
  </versioning>
</metadata>
```

**Pull request closed without merging**